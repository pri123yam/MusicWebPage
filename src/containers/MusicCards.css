.Card{
    height: auto;
    width: 22%;
    display: inline-block;
    border: solid;
    margin: 0.10%;
    position: relative;
    border-color: black;
    border-radius: 18px;
}
.CardImage{
    border-radius: 18px;
    height: 35vh;
    width:100%;
}