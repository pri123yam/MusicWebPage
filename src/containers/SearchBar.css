.Topcomponent{
    background-color: pink;
    text-align: center;
    border-style: solid;
    border-color: palevioletred;
}
.InputBox{
    height: 4vh;
    width: 50%;
    font-size:100%; 
}
.buttonclass{
    height: 4vh;
    width: 10%;
}