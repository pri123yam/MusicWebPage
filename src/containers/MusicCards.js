import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
    Card, CardImg, CardText, CardBody,
    CardTitle, CardSubtitle
} from 'reactstrap';
import './MusicCards.css';
import DisplaySongs from '../selectors/DisplaySongSelector';

export class MusicCards extends Component {
    render() {
        return this.props.songs.map((item) => {
            return (
                <div key={item.id} className='Card'>
                    <Card>
                        <CardImg className='CardImage'  src={item.thumbnail} alt="Card image cap" />
                        <CardBody>
                            <CardTitle>name:{item.name}</CardTitle>
                            <CardSubtitle>genre:{item.genre}</CardSubtitle>
                            <CardText>artist:{item.artist}</CardText>
                        </CardBody>
                    </Card>
                </div>
            );
        });
    }
}
function mapStateToProps(state) {
    return {
        songs: DisplaySongs(state)
    };
}

export default connect(mapStateToProps)(MusicCards);
