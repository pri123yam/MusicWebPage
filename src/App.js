import React, { Component } from 'react';
import './App.css';
import SearchBar from './containers/SearchBar';
import MusicCards from './containers/MusicCards';
import FilterComponent from './containers/Filter';
class App extends Component {
  render() {
    return (
      <div>
        <SearchBar/>
        <FilterComponent/>
        <MusicCards/>
      </div>
    );
  }
}

export default App;