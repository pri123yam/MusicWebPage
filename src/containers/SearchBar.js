import React,{Component} from 'react';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import {searchSongs} from '../actions/SearchSong'
import './SearchBar.css'
export class SearchBar extends Component{
    handleChange=(event)=>{
        var searchKey=event.target.value;
        this.props.searchSongs(searchKey);
    }
    render(){
        return(
            <div className="Topcomponent">
                <h2>MUSIC WEB PAGE</h2>
                <input className='InputBox'
                placeholder='type search text here' type="text"
                onChange={this.handleChange}
                />
            </div>
        )
    }
}

function matchDispatchToProps(dispatch){
    return bindActionCreators({searchSongs},dispatch);
}
export default connect(null,matchDispatchToProps)(SearchBar);