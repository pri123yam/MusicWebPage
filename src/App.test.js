import React from 'react';
import App from './App';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SearchBar from './containers/SearchBar';
import FilterComponent from './containers/filter';
import MusicCards from './containers/MusicCards';
Enzyme.configure({ adapter: new Adapter() });

describe('<App/>',()=>{
    it('should render connected <App/> components correctly',()=>{
        const componentApp=Enzyme.shallow(<App/>);
        expect(componentApp.debug()).toMatchSnapshot();
        expect(componentApp.find(SearchBar)).toHaveLength(1);
        expect(componentApp.find(FilterComponent)).toHaveLength(1);
        expect(componentApp.find(MusicCards)).toHaveLength(1);
    });
});