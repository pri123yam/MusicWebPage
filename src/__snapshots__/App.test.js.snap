// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> should render connected <App/> components correctly 1`] = `
"<div>
  <Connect(SearchBar) />
  <Connect(FilterComponent) />
  <Connect(MusicCards) />
</div>"
`;
